### PR TITLE
Hook up DiagnosticTelemetryPipelineListener in AzureMonitorExporterBuilder

### PR DESCRIPTION
**Pull request by trask**
(this is already being used in the Java agent)

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java [hunk 1]
```diff
@@ -484,22 +484,29 @@ private TelemetryItemExporter createTelemetryItemExporter(StatsbeatModule statsb
...
+                    new DiagnosticTelemetryPipelineListener(
+                        "Sending telemetry to the ingestion service",
+                        true,
+                        " (telemetry will be stored to disk and retried)"),
                     new LocalStorageTelemetryPipelineListener(
                         50, // default to 50MB
                         TempDirs.getSubDir(tempDir, "telemetry"),
                         telemetryPipeline,
-                        LocalStorageStats.noop(),
+                        statsbeatModule.getNonessentialStatsbeat(),
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:506]: we don't collect non-essential statsbeat in the standalone exporter. getNonessentialStatsbeat() will return null.

Comment (trask) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java:506]: ah, thx, I'm still not clear why we don't collect the same in both places, but will revert this change for this PR and we can revisit later

**Comment by heyams** [comment]
we don't. dotnet only collects attach. i asked Ram to have the spec somewhere.

**Comment by heyams** [comment]
do we want to add statsbeat listener?

**Review by heyams — APPROVED** [review]
per our offline discussion, don't need to worry about the statsbeat listener yet.

**Pull request merged**